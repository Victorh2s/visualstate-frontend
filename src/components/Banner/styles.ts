import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css`

  width: 100%;
  height: 22rem;
  overflow: hidden;
  cursor: pointer;


  `}
`;

export const Banner = styled.img`
  ${({ theme }) => css`
  width: 100%;

  `}
`;

export const GladientVertical = styled.div`
  ${({ theme }) => css`


  width: inherit;
  height: inherit;
  background: linear-gradient(to bottom, #0d0c0c 5%, transparent 100%);


  `}
`;

export const DivButtons = styled.div`
  ${({ theme }) => css`

  margin-top: -3rem;
  float: right;
  width: 10%;
  height:3rem;
  border-top-left-radius: 10px;
  background: ${theme.colors.dark};
  text-align: center;
  cursor: pointer;

  > svg {
    color: ${theme.colors.white};
  `}
`;

export const Edit = styled.button`
  ${({ theme }) => css`
  border: none;
  width: 3rem;
  height:3rem;
  background: transparent;
  cursor: pointer;

  > svg {
    color: ${theme.colors.white};
  `}
`;
