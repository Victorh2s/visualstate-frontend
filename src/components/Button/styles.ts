import styled, { css } from 'styled-components';
import { ButtonProps } from '.';

export const Button = styled.button<Pick<ButtonProps, 'color'>>`
  ${({ theme, color }) => css`
    background: ${theme.colors[color]};
    border: none;
    color: ${theme.colors.white};
    font-size: ${theme.font.sizes.small};
    padding: ${theme.spacings.small} ${theme.spacings.medium};
    cursor: pointer;
    border-radius: 1rem;
    transition: all 300ms ease-in-out;
    display: flex;
    align-items: center;
    justify-content: center;

    &:focus {
      outline: none;
      box-shadow: 0 0 ${theme.spacings.small} ${theme.colors[color]};
      filter: brightness(110%);
    }

    &:hover {
      filter: brightness(90%);
    }

    &:disabled {
      background: ${theme.colors.gray8};
      color: ${theme.colors.white};
      cursor: not-allowed;

      &:hover {
        filter: none;
      }
    }

    > svg {
      width: 2rem;
      height: 2rem;
      margin-left: 1rem;
    }
  `}
`;
