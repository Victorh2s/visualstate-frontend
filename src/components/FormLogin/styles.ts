import styled, { css } from 'styled-components';

export const Form = styled.form`
`;

export const ButtonWrapper = styled.div``;

export const Container = styled.div`
    text-align: center;
    width: 35vw;
`;

export const Wrapper = styled.div`
  ${({ theme }) => css`
    width: 100%;
    margin-bottom: ${theme.spacings.large};

  `}
`;

export const InputWrapper = styled.div`
  ${({ theme }) => css`
    position: relative;

    > svg {
      position: absolute;
      top: 50%;
      transform: translate(0, -50%);
      right: 1rem;
      width: 2.5rem;
      height: 2.5rem;
      color: ${theme.colors.gray6};
      z-index: ${theme.layers.layer1};

    }

    *:focus ~ svg {
      color: ${theme.colors.primary};

    }

  `}
`;

export const Input = styled.input`
  ${({ theme }) => css`
    border: 1px solid ${theme.colors.gray3};
    width: 100%;
    height: 100%;
    font-size: ${theme.font.sizes.normal};
    transition: ${theme.transitions.fast};
    padding: ${theme.spacings.small} ${theme.spacings.xsmall};
    background: ${theme.colors.white};
    padding-right: 3.7rem;
    border-radius: ${theme.spacings.tiny};
    outline: none;

    &::placeholder {
      visibility: hidden;
      opacity: 0;
    }

    &:focus {
      border: ${theme.spacings.xxtiny} solid ${theme.colors.primary};
      box-shadow: 0 0 ${theme.spacings.xtiny} 0 ${theme.colors.primary};
      background: ${theme.colors.white};
    }

    &:focus
      + ${Label},
      &:not(:placeholder-shown)
      + ${Label},
      &:-webkit-autofill
      + ${Label} {
      top: 0;
      transform: translate(0, -50%);
      font-size: ${theme.font.sizes.xsmall};
      color: ${theme.colors.white};
      background: ${theme.colors.primary};
      filter: none;
    }


  `}
`;

export const Label = styled.label<{ element: string }>`
  ${({ theme, element }) => css`
    display: inline;
    position: absolute;
    top: 50%;
    transform: translate(0, -50%);
    left: ${theme.spacings.xsmall};
    font-size: ${theme.font.sizes.small};
    height: ${theme.font.sizes.small};
    transition: ${theme.transitions.fast};
    background: ${theme.colors.white};
    padding: 0.5rem 1rem;
    display: flex;
    align-items: center;
    line-height: 1;
    z-index: ${theme.layers.layer1};
    color: ${theme.colors.gray6};
    border-radius: ${theme.spacings.tiny};

    ${
      element === 'textarea' &&
      css`
      top: ${theme.spacings.large};
      transform: translate(0, -50%);
    `
    }
  `}
`;

export const DivButton = styled.div`

`;
export const DivLink = styled.div`
  ${({ theme }) => css`

    margin-bottom: ${theme.spacings.large};
    > a{
      text-decoration: none;
    }
  `}
`;

export const button = styled.button`
  ${({ theme }) => css`

    background: ${theme.colors.primary};
    border: none;
    color: ${theme.colors.white};
    font-size: ${theme.font.sizes.small};
    padding: ${theme.spacings.small} ${theme.spacings.medium};
    cursor: pointer;
    border-radius: 1rem;
    transition: all 300ms ease-in-out;
    display: flex;
    margin: 0 auto;

    &:focus {
      outline: none;
      box-shadow: 0 0 ${theme.spacings.small} ${theme.colors.primary};
      filter: brightness(110%);
    }

    &:hover {
      filter: brightness(90%);
    }



    > svg {
      width: 2rem;
      height: 2rem;
      margin-left: 1rem;
    }
  `}
`;

export const ErrorMessage = styled.p`
  ${({ theme }) => css`
    color: ${theme.colors.warning};
    margin: ${theme.spacings.xxsmall} 0 ${theme.spacings.medium};
  `}
`;
