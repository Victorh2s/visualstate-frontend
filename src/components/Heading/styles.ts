import styled, { css, DefaultTheme } from 'styled-components';
import { HeadingProps } from '.';

const titleSize = {
  supersmall: (theme: DefaultTheme) => css`
  font-size: ${theme.font.sizes.xxsmall};
    ${mediaFont(theme)};
  `,
  small: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.medium};
    ${mediaFont(theme)};
  `,
  medium: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.large};
    ${mediaFont(theme)};
  `,
  big: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.xlarge};
    ${mediaFont(theme)};
  `,
  huge: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.xhuge};
    ${mediaFont(theme)};
  `,
};

const mediaFont = (theme: DefaultTheme) => css`
  @media ${theme.media.lteMedium} {
    font-size: ${theme.font.sizes.medium};
  }
  @media ${theme.media.lteSmall} {
    font-size: ${theme.font.sizes.small};
  }
  @media ${theme.media.lteSuperSmall} {
    font-size: ${theme.font.sizes.xxsmall};
  }
`;

const titleCase = (uppercase: boolean) => css`
  text-transform: ${uppercase ? 'uppercase' : 'none'};
`;

export const Title = styled.h1<HeadingProps>`
  ${({ theme, colorDark, size, uppercase }) => css`
    color: ${colorDark ? theme.colors.dark : theme.colors.white};
    ${titleSize[size](theme)};
    ${titleCase(uppercase)};
  `}
`;
