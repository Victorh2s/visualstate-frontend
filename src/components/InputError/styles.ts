import styled, { css } from 'styled-components';

export const ErrorMessage = styled.p`
  ${({ theme }) => css`
    color: ${theme.colors.warning};
    margin: ${theme.spacings.xxsmall} 0 ${theme.spacings.large};
  `}
`;
