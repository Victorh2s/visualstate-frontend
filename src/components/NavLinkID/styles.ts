import styled, { css } from 'styled-components';

export const Container = styled.nav`
  ${({ theme }) => css`
    display: flex;
    flex-flow: row wrap;
    align-items: center;
    justify-content: center;
    @media ${theme.media.lteMedium} {
      flex-flow: column wrap;
      align-content: center;
    }

  `}
`;
export const Logo = styled.img`
  ${({ theme }) => css`
    width: 6rem;
    border-radius: 5rem;
    padding: ${theme.spacings.small} ;

  `}
`;
export const User = styled.div`
  ${({ theme }) => css`
    border-radius: 5rem;
    font-size: ${theme.font.sizes.xsmall};
    color: ${theme.colors.white};
  `}
`;

export const Welcome = styled.div`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.xsmall};
  `}
`;

export const DivUser = styled.div`
  ${({ theme }) => css`
  display: flex;
  align-items: center;
    margin-right: 10rem;
    margin-left: -1rem;
    @media ${theme.media.lteMedium} {
      display: block;
      margin-right: 0;
      margin-left: 0;
      margin-bottom: 5rem;
    }
  `}
`;

export const Logout = styled.button`
  ${({ theme }) => css`
    background: transparent;
    border: none;
    cursor: pointer;

  > svg {
      width: 2rem;
      height: 2rem;
      color: ${theme.colors.white};
    }
  `}
`;
