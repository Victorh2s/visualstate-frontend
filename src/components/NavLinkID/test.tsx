import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import { NavLinksID } from '.';
import { theme } from '../../styles/theme';

describe('<NavLinksID />', () => {
  it('should render links', () => {
    renderTheme(<NavLinksID />);
    expect(screen.getByLabelText('Main menu')).toBeInTheDocument();
    expect(screen.getByRole('img')).toBeInTheDocument();
  });

  it('should not render links', () => {
    renderTheme(<NavLinksID />);
    expect(screen.queryAllByText(/links/i)).toHaveLength(0);
  });

  it('should match snapshot', () => {
    const { container } = renderTheme(<NavLinksID />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
