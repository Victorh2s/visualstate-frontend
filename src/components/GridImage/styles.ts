import styled, { css } from 'styled-components';
import { Title as HeadingContainer } from '../Heading/styles';
import { Container as TextComponent } from '../TextComponent/styles';

export const Container = styled.div`
${({ theme }) => css`
padding: 2rem;
padding-bottom: 10rem;
 > ${TextComponent}{
  text-align: center;
  margin-bottom: ${theme.spacings.xhuge}
}
${HeadingContainer}{
  text-align: center;
}
`}
`;

export const Grid = styled.div`
${({ theme }) => css`
  display: grid;
  grid-template-columns: repeat(auto-fill,minmax(280px, 1fr));
  gap: ${theme.spacings.large}
`}
`;

export const GridElement = styled.div`
  overflow: hidden;
`;

export const Image = styled.img`
  width: 100%;
  transition: all 300ms ease-in-out;
  &:hover{
    transform: scale(1.2);
  }
`;
