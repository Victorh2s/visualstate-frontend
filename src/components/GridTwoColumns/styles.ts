import styled, { css } from 'styled-components';
import { Title } from '../Heading/styles';

export type translatey = {
  translatey: number;
};

export const Container = styled.div`
  ${({ theme }) => css`
    display: flex;
    overflow-x: hidden;
    align-items: center;
    width: 100%;
    gap: ${theme.spacings.large};
    @media ${theme.media.lteOrEqMedium} {
      display: block;
      text-align: center;
      align-items: center;
      overflow-x: hidden;
    }
    @media ${theme.media.lteOrEqSmallest} {
      width: 70%;
  }
    ${Title} {
      margin-bottom: 0;
    }
  `}
`;

export const TextContainer = styled.div`
    width: 35rem;
    display: flex;
    overflow-x: hidden;
    margin: 0 auto;
    padding-bottom: 0.1rem;
`;

export const Ytext = styled.div<translatey>`
  ${({ theme, translatey }) => css`
    display: flex;
    transform: translateX(-${translatey}rem);
    transition: all 400ms ease-in-out;
    padding-bottom: 0.1rem;
    @media ${theme.media.lteOrEqSmallest} {
      transform: translateX(-${translatey - 11}rem);
      transition: all 400ms ease-in-out;
    }
`}
`;

export const HeadingHeight = styled.div`
 ${({ theme }) => css`
  width: 35rem;
  padding-bottom: 0.1rem;
  @media ${theme.media.lteOrEqSmallest} {
      width: 25rem;
  }
  `}
`;

export const Carrousel = styled.div`
 ${({ theme }) => css`
  height: 40rem;
  margin: 0 auto;
  overflow-y: hidden;
  @media ${theme.media.lteOrEqSmallest} {
    height: 25rem;
  }

  `}
`;

export const Yvideo = styled.div<translatey>`
  ${({ theme, translatey }) => css`
    transform: translateY(-${translatey}rem);
    transition: all 400ms ease-in-out;
    @media ${theme.media.lteOrEqMedium} {
      transform: translateY(-${translatey + 12}rem);
      transition: all 400ms ease-in-out;
  }

  `}
`;

export const SourceContainer = styled.div`
  width: 100%;
  height: 50rem;
`;

export const videos = styled.video`
  ${({ theme }) => css`
  width: 45vw;
  height: 50rem;
  @media ${theme.media.lteOrEqMedium} {
    width: 80%;
    height: 50rem;
    align-items: center;
  }
  @media ${theme.media.lteOrEqSmallest} {
    width: 60%;
    height: 50rem;
    align-items: center;
  }
    `}
`;
