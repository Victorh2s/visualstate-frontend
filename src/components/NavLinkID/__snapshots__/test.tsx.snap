// Vitest Snapshot v1

exports[`<NavLinksID /> > should match snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c2 {
  width: 6rem;
  border-radius: 5rem;
  padding: 1.6rem;
}

.c3 {
  border-radius: 5rem;
  font-size: 1.2rem;
  color: #FFFFFF;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-right: 10rem;
  margin-left: -1rem;
}

.c4 {
  background: transparent;
  border: none;
  cursor: pointer;
}

.c4 > svg {
  width: 2rem;
  height: 2rem;
  color: #FFFFFF;
}

.c5 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

@media (max-width:768px) {
  .c0 {
    -webkit-flex-flow: column wrap;
    -ms-flex-flow: column wrap;
    flex-flow: column wrap;
    -webkit-align-content: center;
    -ms-flex-line-pack: center;
    align-content: center;
  }
}

@media (max-width:768px) {
  .c1 {
    display: block;
    margin-right: 0;
    margin-left: 0;
    margin-bottom: 5rem;
  }
}

<nav
  aria-label="Main menu"
  class="c0"
>
  <div
    class="c1"
  >
    <img
      class="c2"
    />
    <div
      class="c3"
    />
  </div>
  <button
    class="c4"
  >
    <svg
      aria-hidden="true"
      class="c5"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0V0z"
        fill="none"
      />
      <path
        d="M17 8l-1.41 1.41L17.17 11H9v2h8.17l-1.58 1.58L17 16l4-4-4-4zM5 5h7V3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h7v-2H5V5z"
      />
    </svg>
  </button>
</nav>
`;
